lib/ansible/inventory/manager.py

```diff
@@ -118,12 +118,17 @@ def split_host_pattern(pattern):
 class InventoryManager(object):
     ''' Creates and manages inventory '''
 
-    def __init__(self, loader, sources=None):
+    def __init__(self, loader, sources=None, paths=None):
```

Comment:
I'm conflicted on this, as then we are tying the playbook paths to inventory, and the naming here is ambiguous.

I think I'd be happier if we could try to divorce these more.  I realize that we don't have `inventory_loader` in `cli`, and nor should we, and we need those dirs added before we search for inventory plugins.

Maybe we could create a top level function in `inventory/manager.py` that can do this that we can import?  Rather than tying it directly to the `InventoryManager` class where it really isn't needed.

Reply:
I suppose my feedback is just that `InventoryManager` doesn't actually need those paths, and does nothing with them after adding dirs to `inventory_loader`.  As such, I think we should move that work out of the class.

Maybe just create something like the following as a top level func in `inventory/manager.py`:

```
def add_inventory_loader_dirs(play_paths):
    if play_paths is not None:
         for path in play_paths:
             inventory_loader.add_directory(path, with_subdir=True)
```

...and in `cli/__init__.py` import that function, and call it independently of `InventoryManager`. 

This way we aren't passing unneeded info to `InventoryManager`, we aren't importing `inventory_loader` into cli, and we've still grouped code related to inventory in `manager.py`.

Otherwise, maybe we just import `inventory_loader` into `cli` and do it directly there.